### PR TITLE
[CARBONDATA-2363][branch-1.3] Add CarbonStreamingQueryListener to SparkSession

### PR DESCRIPTION
**Pull request by QiangCai**
Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [x] Any interfaces changed?
 
 - [x] Any backward compatibility impacted?
 no
 - [x] Document update required?
no
 - [x] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [x] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. 
small

### CODE REVIEW

File: streaming/src/main/scala/org/apache/carbondata/streaming/StreamSinkFactory.scala [hunk 1]
```diff
@@ -77,6 +78,22 @@ object StreamSinkFactory {
     }
   }
 
+  lazy val listenerAdded = new mutable.HashMap[Int, Boolean]()
+
+  // add CarbonStreamingQueryListener to other SparkSession
```
Comment (jackylk) [inline streaming/src/main/scala/org/apache/carbondata/streaming/StreamSinkFactory.scala]: please change to `add CarbonStreamingQueryListener if it is SparkSession instance`

File: streaming/src/main/scala/org/apache/carbondata/streaming/StreamSinkFactory.scala [hunk 2]
```diff
@@ -77,6 +78,22 @@ object StreamSinkFactory {
...
+  def addListenerForSparkSession(sparkSession: SparkSession): Unit = {
+    if (!"CarbonSession".equals(sparkSession.getClass.getSimpleName)) {
```
[the remark this answers is not in the document]
Comment (QiangCai) [reply streaming/src/main/scala/org/apache/carbondata/streaming/StreamSinkFactory.scala]: ok

**Comment by jackylk** [comment]
LGTM

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5215/

**Comment by zzcclp** [comment]
@jackylk @QiangCai I raise a hotfix pr to fix import error after this pr was merged. The 'CarbonStreamingQueryListener' is in the package 'import org.apache.spark.sql.execution.streaming' on branch-1.3, not the package 'org.apache.carbondata.streaming'.

**Pull request closed without merging**